core/src/main/scala/cr/pulsar/Topic.scala

```diff
@@ -105,28 +112,30 @@ object Topic {
       this.copy(_type = typ)
 
     /**
-      * It creates a topic. By default, Type=Persistent.
+      * It creates a topic of type Single. By default, Type=Persistent.
       */
     def build(
         implicit @implicitNotFound(
           "Topic.Name and Config are mandatory. By default Type=Persistent."
-        ) ev: I =:= Info.Mandatory
-    ): Topic =
-      new Topic {
-        val name = _name
-        val url  = buildUrl(_config, _name, _type)
+        ) ev: I =:= Info.SingleMandatory
+    ): Topic.Single = {
+      val t = _name.swap.toOption.get
```

Comment:
There are 2 places where we call `Option.get`. Is there a way to get rid of it? It's a potentially dangerous operation that might cause issues later on when we will change something in the class definition. 
Also, having 2 different constructors for different topic types seems to be confusing as you have to keep in mind what kind of topic you've put into the method

Reply:
This is safe because it's validated at the type-level. Users don't need to keep in mind whether they used `withName` or `withNamePattern` because this is validated at *compile time*, that's the cool thing about using a type-level builder :)

Reply:
I see. That makes sense, but isn't it simpler just to have one `build` method:
```
def build = topic match {
  case x: Name => new Single {} 
  case y: NamePattern => new Multi {}
}
```

Reply:
Unfortunately, we need the return type to be one of the branches, no the parent class. This is because only consumers support the `Multi` topic, readers and producers only support `Single`.

Reply:
Aha, that sounds like a good reason

Reply:
Yeah I went full circle with the entire design, trust me 😄